core/app_client.go

```diff
@@ -237,8 +237,16 @@ func (a appClient) GetTopAndLocalScores(orgID string, appID string, userID strin
 	userScore := scores[0]
 	scores = scores[1:]
 
-	// Insert user's score into section with surrounding local scores
-	if len(scores) > *limit {
+	userInTopScoresIdx := -1
+	for i, score := range scores {
+		if score.UserID == userID {
+			userInTopScoresIdx = i
+			break
+		}
+	}
+
+	// Insert user's score into section with surrounding local scores if user is not in top scores
+	if len(scores) > *limit && userInTopScoresIdx == -1 {
```

Comment:
Doesn't this change prevent us from handling the local scores properly? In other words, without this isn't there an issue where all equal scores will be included rather than just a couple above and below?